### DISCUSSION TITLE
WSL2 not loading custom kernel from kernel.org

### DISCUSSION THREAD

**Discussion by kuyacarlo** [microsoft/WSL#8744] [General] [answered]
# Environment

```
Windows build number: 22000.856
Your Distribution version: 20.04
uname -a: Linux user 5.10.102.1-microsoft-standard-WSL2 #1 SMP Wed Mar 2 00:30:59 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```
# Steps to reproduce
1. Get the kernel from https://kernel.org, I get 518 as it is the stable version.
2. Build and use via .wslconfig:
```
[wsl2]
kernel=C:\Users\sjc31\wsl\kernel\vmlinux\5.18.18-wsl-x86_64-mainline
```

# Expected behavior

uname -ar lists the kernel wich i build, instead I see the default one.

# Actual behavior

5.10.102.1-microsoft-standard-WSL2 the default one and not what I compiled.

# Additional Information:
I used this guide: https://michaelschaecher.github.io/2022-01-15-how-to-compile-the-kernel-for-wsl/
Also, I did #6156 comment suggesting to use the vmlinux file but no luck.

**Comment by NotTheDr01ds** [comment, marked as answer]
Try:

```
[wsl2]
kernel=C:\\Users\\sjc31\\wsl\\kernel\\vmlinux\\5.18.18-wsl-x86_64-mainline
```

Note that the blog post you linked to did use escaped/double-backslashes, but it's easy to miss.  I've forgotten to escape them multiple times myself.

**Reply by kuyacarlo** [reply]
It worked.

Additional Note:
You should place the path of the kernel, not the folder. Linking the folder may leave you with access denied.
